omicsclaw/memory/compat.py

```
@@ -357,25 +444,38 @@ async def _collect(uri: str, mtype: str):
+
+        Searches every cached per-namespace client; without a session_id
+        the caller hasn't told us which user the memory_id belongs to,
+        and we don't want a global scan that could match the wrong user's
+        memory_id collision. ``memory_id`` is a uuid4 so collisions are
+        astronomically unlikely, but the per-namespace search keeps the
+        update strictly within whichever namespace the row lives.
+        """
+        # Touch the session client first so the shared partition is
+        # always considered (covers the legacy "everything in shared"
+        # case before the namespace migration).
+        candidate_clients = [self._session_client] + list(self._memory_clients.values())
```

Ruff (0.15.12)

[warning] Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)